src/engine/tests/Test_DisjunctionConstraint.h

```diff
@@ -500,4 +501,141 @@ class DisjunctionConstraintTestSuite : public CxxTest::TestSuite
         TS_ASSERT( !dc.isFeasible() );
         TS_ASSERT_EQUALS( dc.nextFeasibleCase(), CONSTRAINT_INFEASIBLE );
     }
+
+    void test_disjunction_make_all_disjuncts_bounds()
+    {
+        InputQuery inputQuery;
+
+        PiecewiseLinearCaseSplit cs4;
+
+        // x1 - x0 <= 1, x1 + x2 >= 2
+        Equation eq1;
+        eq1._type = Equation::LE;
+        eq1.addAddend( 1, 1 );
+        eq1.addAddend( -1, 0 );
+        eq1.setScalar( 1 );
+        cs4.addEquation( eq1 );
+        Equation eq2;
+        eq2._type = Equation::GE;
+        eq2.addAddend( 1, 1 );
+        eq2.addAddend( 1, 2 );
+        eq2.setScalar( 2 );
+        cs4.addEquation( eq2 );
+
+        List<PiecewiseLinearCaseSplit> caseSplits = { *cs2, *cs3, cs4 };
+        DisjunctionConstraint *disj = new DisjunctionConstraint( caseSplits );
+
+        inputQuery.setNumberOfVariables( 3 );
+        inputQuery.addPiecewiseLinearConstraint( disj );
+
+        List<PhaseStatus> casesBefore = disj->getAllCases();
+        List<PiecewiseLinearCaseSplit> splitsBefore = disj->getCaseSplits();
+        TS_ASSERT_THROWS_NOTHING( disj->transformIfNeeded( inputQuery ) );
+        List<PhaseStatus> casesAfter = disj->getAllCases();
+        List<PiecewiseLinearCaseSplit> splitsAfter = disj->getCaseSplits();
+
+        TS_ASSERT_EQUALS( casesBefore.size(), casesAfter.size() );
+        TS_ASSERT_EQUALS( splitsBefore.size(), splitsAfter.size() );
+
+        // 1 <= x0 <= 5, x1 = x0
+        // 5 <= x0 , x1 = 2x2 + 5
+        // x1 - x0 <= 1, x1 + x2 >= 2
+
+        // In total there are 4 (in)equalities in all the disjuncts.
+        TS_ASSERT_EQUALS( inputQuery.getNumberOfVariables(), 9u );
+        TS_ASSERT_EQUALS( inputQuery.getEquations().size(), 6u );
```

Comment:
Should these assertions have their before version, to highlight what is the difference (if any) during the transformation?

Reply:
Will add the before assertions.

Reply:
done.